sql-plugin/src/main/java/com/nvidia/spark/rapids/GpuColumnVector.java

```diff
@@ -629,6 +629,17 @@ public static ColumnarBatch from(Table table, DataType[] colTypes) {
    */
   static boolean typeConversionAllowed(ColumnView cv, DataType colType) {
     DType dt = cv.getType();
+    // We should honor the actual decimal type used for the cudf column first, then the precision.
+    // Instead of always inferring a deciaml type from the precision of Spark. Because both
+    // DECIMAL32 and DECIMAL64 can support precisions <= MAX_INT_DIGITS. Cudf may use either one
+    // for this case. e.g. ORC reader will always read decimal as DECIMAL64, even the precision
```

Comment:
Should the plugin's ORC reader be downcasting columns to `DECIMAL32` in the case of excessively-precise decimals?  That could save space on operations downstream from the query.

Reply:
Yes, it will be better to cast columns to `DECIMAL32` when the precision allows. However I think this is an improvement since it is not required for the functionality. 

I thought this for a while. Seems we need to iterate the columns (including nested columns) to find out the ones matching the case and do the casting. And compond columns need to be rebuilt with new cast children. The whole process looks like a rule for space optimization, which can be used in more places than just the ORC reader.
How about implementing it as an util method, instead of integrating it with ORC reader in another PR ?

Reply:
I think we should be consistent with how the decimals are loaded.  The Parquet reader is already converting DECIMAL64 columns into DECIMAL32 columns on read when appropriate, so I think we should do the same here.  The code to convert is already written and can be refactored from the Parquet code as a utility method.

Reply:
The Parquet code for this is part of `ParquetPartitionReaderBase#evolveSchemaIfNeededAndClose`

Reply:
Got it. Updated.